BeagleBone_black/src/state_machine/states.hpp

```diff
@@ -48,6 +51,8 @@ class State {
   virtual void entry() = 0;
   data::State   state_;
   static State* alloc_;   // allocate all states here
+  GPIO pin_37_ = GPIO(78, utils::io::gpio::kOut);
```

Comment:
this is not good,, if you look at how the state machine transitions between states, it always creates new states, thus the construction of these gpios would be constructed every time we transition to another state. Make it more global to have a single allocation, e.g. just add it to the top of .cpp file, plus, drive them high right after the construction,, it is too late to do it in the idle state

Reply:
and while you are at it, remove the system member variable from state.hpp, in particular from Idle and EmergencyBraking states.